web/modules/custom/dpl_fbi/src/Plugin/Field/FieldWidget/WorkIdSearchForMaterialWidget.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\dpl_fbi\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\dpl_react_apps\Controller\DplReactAppsController;
+
+/**
+ * Defines the 'dpl_fbi_work_id_search_for_material' field widget.
+ *
+ * @FieldWidget(
+ *   id = "dpl_fbi_work_id_search_for_material",
+ *   label = @Translation("Work ID search for material"),
+ *   field_types = {"dpl_fbi_work_id"},
+ * )
+ */
+final class WorkIdSearchForMaterialWidget extends WidgetBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state): array {
+
+    $uniqId = uniqid((string) rand(), TRUE);
+    $identifier = "{$uniqId}-{$delta}";
```

Comment:
Please consider this approach.

I understand the intention to generate a unique identifier for the field - and I am pretty sure that this will be the case for the current implementation - but is seens as if you misunderstand parts of what the different elements do.

[`uniqid()`](https://www.php.net/manual/en/function.uniqid.php) generateres a unique id based on the current time in microsecords. You then use a random value for the prefix.

The description of the `prefix` argument is to separate different ids which might be generated at the same microsecond - e.g. across servers or the like. 

You then append the delta - the numeric instance of the field if it has multiple values.

As I see it we only need to ensure uniqueness on a single page/form.

You already have the delta. You can get a unique identifier for the field through `FieldItemListInterface->getFieldDefinition()->getUniqueIdentifier()`. I would think that you would get something sensible by combining the two?

Reply:
I see. I didn't realise `FieldItemListInterface->getFieldDefinition()->getUniqueIdentifier()` was available. have updated 👍